src/emcpy/plots/create_plots.py

```diff
@@ -1,963 +1,717 @@
-                    method=plotobj.interp, bounds_error=False)
+                    method=plotobj.density['interp'], bounds_error=False)
         # To be sure to plot all data
         z[np.where(np.isnan(z))] = 0.0
         # Sort the points by density, so that the densest
         # points are plotted last
-        if plotobj.sort:
+        if plotobj.density['sort']:
             idx = z.argsort()
             x, y, z = plotobj.x[idx], plotobj.y[idx], z[idx]
-        axis = self._determine_axis(plotobj.plot_ax)
-        cs = axis.scatter(x, y, c=z,
-                          s=plotobj.markersize,
-                          cmap=plotobj.cmap,
-                          label=plotobj.label)
+        cs = ax.scatter(x, y, c=z,
+                        s=plotobj.markersize,
+                        cmap=plotobj.density['cmap'],
+                        label=plotobj.label)
         # below doing nothing? fix/remove in subsequent PR?
         # norm = Normalize(vmin=np.min(z), vmax=np.max(z))
 
-        if plotobj.colorbar:
+        if plotobj.density['colorbar']:
             self.cs = cs
 
-    def _scatter(self, plotobj):
+    def _scatter(self, plotobj, ax):
         """
         Uses Scatter object to plot on axis.
         """
-        axis = self._determine_axis(plotobj.plot_ax)
-
         # checks to see if density attribute is True
-        if plotobj.density:
-            self._density_scatter(plotobj)
-
+        if hasattr(plotobj, 'density'):
+            self._density_scatter(plotobj, ax)
         else:
-            s = axis.scatter(plotobj.x, plotobj.y,
-                             s=plotobj.markersize,
-                             color=plotobj.color,
-                             marker=plotobj.marker,
-                             vmin=plotobj.vmin,
-                             vmax=plotobj.vmax,
-                             alpha=plotobj.alpha,
-                             linewidths=plotobj.linewidths,
-                             edgecolors=plotobj.edgecolors,
-                             label=plotobj.label)
-
-        # checks to see if linear regression attribute is True
-        if plotobj.linear_regression:
-            y_pred, r_sq, intercept, slope = get_linear_regression(plotobj.x,
-                                                                   plotobj.y)
-            label = f"y = {slope:.4f}x + {intercept:.4f}\nR\u00b2 : {r_sq:.4f}"
-            axis.plot(plotobj.x, y_pred, color=plotobj.lr_color,
-                      linewidth=plotobj.lr_linewidth, label=label)
-
-    def _histogram(self, plotobj):
+            skipvars = ['plottype', 'plot_ax', 'x', 'y',
+                        'markersize', 'linear_regression',
+                        'density', 'channel']
+            inputs = self._get_inputs_dict(skipvars, plotobj)
+            s = ax.scatter(plotobj.x, plotobj.y, s=plotobj.markersize,
+                           **inputs)
+
+        # checks to see if linear regression attribute
```

Comment:
Very minor -- the comment isn't entirely clear.  Maybe "checks to see if linear regression attribute exists" or something similar.